modules/mauth-common/src/main/java/com/mdsol/mauth/MAuthRequest.java

```diff
@@ -85,21 +119,29 @@ public MAuthRequest(String authenticationHeaderValue, byte[] messagePayload, Str
     validateNotBlank(httpMethod, "Http method");
     validateNotBlank(resourcePath, "Resource path");
     validateRequestTime(requestTime);
-    if (messagePayload == null) {
-      messagePayload = new byte[] {};
-    }
     if (queryParameters == null) {
       queryParameters = "";
     }
 
     this.appUUID = appUUID;
     this.requestSignature = requestSignature;
-    this.messagePayload = messagePayload;
     this.httpMethod = httpMethod;
     this.requestTime = requestTime;
     this.resourcePath = resourcePath;
     this.queryParameters = queryParameters;
     this.mauthVersion = MAuthHeadersHelper.getMauthVersion(authenticationHeaderValue);
+
+    // we always use inputStream for Auth,
+    // wrap byte array in ByteArrayInputStream if bodyInputStream wasn't provided explicitly
+    validatePayload(messagePayload, bodyInputStream);
```

Comment:
I don't think extracting this out helps readability. We want to inline it because it's only used in one place, and having the `if` conditions here makes it very clear what was checked 

Or put it another way: If you're a new reader reading this code, do you know what is checked by just reading this line and how it affects our assumption about the logic in the following lines?

Reply:
updated in https://github.com/mdsol/mauth-jvm-clients/pull/132/commits/ac9b72d9407033a365bbc428dfdd2315ead45826